rasa/nlu/config.py

```diff
@@ -76,9 +77,21 @@ def component_config_from_pipeline(
     pipeline: List[Dict[Text, Any]],
     defaults: Optional[Dict[Text, Any]] = None,
 ) -> Dict[Text, Any]:
+    """Get the configuration of the `index`th component.
+
+    Args:
+        index: Index of the component.
+        pipeline: List of component configurations.
```

Comment:
I'd avoid repeating the type. 
```suggestion
        pipeline: Configuration of the components.
```